src/main/java/com/rakovets/course/java/core/practice/looping_statements/Task04.java

```diff
@@ -33,8 +33,20 @@ public static void main(String[] args) {
      * @return время для убийства RaidBoss (когда party не справляется за 24 часа, то вывести -1)
      */
     static int calculateRaidTime(int healthPoints, double regenerationPercentPerHour, int averageDamagePerHour) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int hourInDay = 24;
+        int currentHp = healthPoints;
+        int time = 0;
+        int finalTime;
+            do {
+                currentHp += currentHp * regenerationPercentPerHour / 100;
+                currentHp = currentHp - averageDamagePerHour;
+                time++;
+            } while (currentHp > 0 && time < hourInDay);
+            if (currentHp > 0) {
+                finalTime = -1;
+            } else {
+                finalTime = time;
+            }
```

Comment:
"Violation: There aren't correct indentations for inner code
Solution: Add correct indentations for inner code"